CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -85,9 +85,31 @@ static function X2DataTemplate CreatePromotionsRewardTemplate()
 	// This is a dummy reward, does nothing
 	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_Promotions');
 
+	Template.IsRewardAvailableFn = IsPromotionsRewardAvailable;
+
 	return Template;
 }
 
+static function bool IsPromotionsRewardAvailable(optional XComGameState NewGameState, optional StateObjectReference AuxRef)
+{
+	local X2CovertActionTemplate ActionTemplate;
+	local X2StrategyElementTemplateManager TemplateManager;
+
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	ActionTemplate = X2CovertActionTemplate(TemplateManager.FindStrategyElementTemplate('CovertAction_ExhaustiveTraining'));
+
+	if (ActionTemplate != none)
+	{
+		return ActionTemplate.Slots[0].Rewards.Length > 0;
+	}
+	else
```

Comment:
`else` is redundant (in case `if (ActionTemplate != none)` is matched, the function will `return` anyway).

Fix and merge